_posts/2022-02-25-newpipe-0.22.1.md

```diff
@@ -0,0 +1,52 @@
+---
+layout: post
+title: "NewPipe 0.22.1 released: Fixes and adjustments"
+short: "NewPipe 0.22.1 released"
+date: 2022-02-25 22:00:00 + 01:00
+categories: [pinned, release]
+author: <a href="https://github.com/Stypox">@Stypox</a>
+image: newpipe
+excerpt_separator: <!-- more -->
+---
+
+Why is there a new release only six days after the previous one? Well, don't expect anything big… just a few adjustments to the previous version and some long-awaited bugfixes for lists! That's what you would expect from a point release, according to [semantic versioning](https://semver.org/), but we have never been that good at choosing release numbers, sorry about that.
+
+<!-- more -->
+
+### Fixed
+
+- Lists on big screen might finally work as expected! Now if the items loaded initially do not fill the whole screen, NewPipe will load another chunk of them, instead of showing a loading indicator **forever**. This long-awaited fix was brought to you by none other than the hero from last release, [@litetex](https://github.com/litetex). In the same pull request he also fixed some recurring crashes when scrolling through lists, caused by a misuse of the nasty Android `RecyclerView`. 
+  > For those of you wondering, a `RecyclerView` is a special *view* component that allows showing many *items* of the same type in a list without overloading the system with too many views at the same time. It does so by binding only the items visible on the screen to the corresponding view component (for example a stream info item view), and when one such view is moved out on one side of the screen it is *recycled* to be used again once needed on the opposite side. The random crash that was fixed here was caused a piece of code that ~~sometimes~~ *once in a blue moon* tried to bind the same view component to multiple items at the same time, which is obviously not possible, hence the crash. Sorry for this out-of-section nerd talk, just wanted to communicate why it took us so long to find out the root cause of the problem.
+
+- [@Stypox](https://github.com/Stypox) (I declare freedom from any conflict of interest) solved the graphical problem with the newly-introduced fast-forward/rewind arc not underlaying the system ui when playing in the fullscreen player. This was particularly visible when controls were hidden, since a non-animated border was shown around the animated arc.
```

Comment:
```suggestion
- [@Stypox](https://github.com/Stypox) (I, as the blog author, declare freedom from any conflict of interest) solved the graphical problem with the newly-introduced fast-forward/rewind arc not underlaying the system ui when playing in the fullscreen player. This was particularly visible when controls were hidden, since a non-animated border was shown around the animated arc. This was a really cool fix, very well done [@Stypox](https://github.com/Stypox).
```

Reply:
Tried to make sure who I was in this context (normally we is used when referring to the blog authors) + I played on the conflict of interest joke a bit more, maybe thats too much though?

Reply:
I'd prefer not to include it ;-)